firebase-functions/functions/src/index.ts

```diff
@@ -0,0 +1,22 @@
+import * as functions from 'firebase-functions';
+
+// // Start writing Firebase Functions
+// // https://firebase.google.com/docs/functions/typescript
+//
+// export const helloWorld = functions.https.onRequest((request, response) => {
+//  response.send("Hello from Firebase!");
+// });
+
+const admin = require('firebase-admin');
+admin.initializeApp();
+
+/**
+ * This database triggered function will delete /shared-tables/{shareId} when
+ * /shared-tables/{shareId}/connectedUsers is deleted, which happens after the last person leaves the sahe
+ */
+exports.deleteDataWithNoConnectedUsers = functions.database.ref('/shared-tables/{shareId}/connectedUsers').onDelete(async (change) => {
+  const ref = change.ref.parent;
+  if (ref) {
+    return ref.remove();
+  }
+});
```

Comment:
This seems like the right place to start, but I wonder if we might end up wanting to add a delay of some sort to deal with network hiccups, i.e. allow reconnection to the shared table within a limited time period.

Reply:
Happy to keep thinking about that. From a naive perspective, either someone else is connected to the table, in which case the table won't delete unless all have a hiccup simultaneously, or you're the only one connected. If you're the only one connected, your disconnect will instantly clear the `connectedUsers` array, and our current joining code does not allow joining an existing table unless there is another user in the `connectedUsers` array. So you'd be locked out anyway.